File: Tutorial5-1TableRelationsOneToMany/src/main/java/com/test/tutorial5_1tablerelations/data/UserAndAllPets.java
package com.test.tutorial5_1tablerelations.data;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Relation;

import java.util.List;

/**
 * @Embedded tag is for having nested fields.
 *
 * <p></p>
 * @Relation is for having relation with other model class.
 * Those two parameters say that parentColumn name from User class is id
 * and entityColumn name from Repo class is userId.
 */
public class UserAndAllPets {

    @Embedded
    public User user;

    @Relation(parentColumn = "id", entityColumn = "userId", entity = Pet.class)
    public List<Pet> pets;
}

File: Tutorial5-1RelationsOneToMany/src/main/java/com/test/tutorial5_1tablerelations/data/UserAndAllPets.kt
package com.test.tutorial5_1tablerelations.data


import androidx.room.Embedded
import androidx.room.Relation

/**
 * @Embedded tag is for having nested fields.
 *
 *
 *
 * @Relation is for having relation with other model class.
 * Those two parameters say that parentColumn name from User class is petId
 * and entityColumn name from Repo class is userId.
 */
class UserAndAllPets {

    @Embedded
    var user: User? = null

    // 🔥 'id' comes from User, 'userId' comes from Pet. Both are the same ids
    @Relation(parentColumn = "id", entityColumn = "userId", entity = Pet::class)
    var pets: List<Pet>? = null
}
